src/components/AllSpecies/AllSpecies.tsx

```diff
@@ -8,16 +8,64 @@ type AllSpeciesProps = {
   characters: ModifiedCharacter[]
 }
 
+type CharacterState = {
+  originalCharacters: ModifiedCharacter[]
+  modifiedCharacters: ModifiedCharacter[]
+  selectedValue: string
+}
+
+type Action = { type: 'filter'; event: CharacterState['selectedValue'] }
+
+type Key = 'mass' | 'height'
```

Comment:
[nitpick] - I'm not really impressed by this type's name as `Key` is way too vague. It could be something like; `sortKey` or `sortFactor` or something else.